internal/schema/diff_test.go

```diff
@@ -260,32 +287,115 @@ func TestGetResourcesMap(t *testing.T) {
+		name           string
+		expected       *CostComponent
+	}{
+		{
+			costComponents: []*CostComponent{
+				{Name: "Instance usage"},
+				{Name: "EBS-optimized usage"},
+				{Name: "CPU Credits"},
+			},
+			name: "EBS-optimized usage",
+			expected: &CostComponent{
+				Name: "EBS-optimized usage",
+			},
+		},
+		{
+			costComponents: []*CostComponent{
+				{Name: "Instance usage"},
+				{Name: "EBS-optimized usage"},
+				{Name: "CPU Credits"},
+			},
+			name:     "Not existent",
+			expected: nil,
+		},
+		{
+			costComponents: []*CostComponent{
+				{Name: "Capacity (first 50TB)"},
+				{Name: "Capacity (next 450TB)"},
+				{Name: "Capacity (over 500TB)"},
+			},
+			name: "Capacity (over 500TB)",
+			expected: &CostComponent{
+				Name: "Capacity (over 500TB)",
+			},
+		},
+		{
+			costComponents: []*CostComponent{
+				{Name: "Instance usage (Linux/UNIX, on-demand, t3.small)"},
+				{Name: "EBS-optimized usage"},
+				{Name: "CPU Credits"},
+			},
+			name: "Instance usage (Linux/UNIX, reserved, t3.small)",
+			expected: &CostComponent{
+				Name: "Instance usage (Linux/UNIX, on-demand, t3.small)",
+			},
+		},
+		{
+			costComponents: []*CostComponent{
+				{Name: "Instance usage"},
+				{Name: "EBS-optimized usage"},
+				{Name: "CPU Credits"},
+			},
+			name:     "Instance usage (Linux/UNIX, reserved, t3.small)",
+			expected: nil,
+		},
 	}
-	ccMap := getCostComponentsMap(resource)
-	expectedMap := map[string]*CostComponent{
-		"cc1": cc1,
-		"cc2": cc2,
+
+	for _, test := range tests {
+		actual := findMatchingCostComponent(test.costComponents, test.name)
+		assert.Equal(t, test.expected, actual)
 	}
-	assert.Equal(t, expectedMap, ccMap)
 }
 
-func TestDiffResourcesByKey_bothNil(t *testing.T) {
-	emptyRMap := make(map[string]*Resource)
-	changed, _ := diffResourcesByKey("random_resource", emptyRMap, emptyRMap)
-	assert.Equal(t, false, changed)
-}
+func TestDiffName(t *testing.T) {
+	tests := []struct {
+		current  string
+		past     string
+		expected string
+	}{
+		{
+			current:  "Instance usage",
+			past:     "Instance usage",
+			expected: "Instance usage",
+		},
+		{
+			current:  "Instance usage",
+			past:     "",
+			expected: "Instance usage",
+		},
+		{
+			current:  "",
+			past:     "Instance usage",
+			expected: "Instance usage",
+		},
+		{
+			current:  "Instance usage",
+			past:     "Different",
+			expected: "Instance usage",
+		},
+		{
+			current:  "Instance usage (Linux/UNIX)",
+			past:     "Instance usage (Linux/UNIX)",
+			expected: "Instance usage (Linux/UNIX)",
+		},
+		{
+			current:  "Instance usage (Linux/UNIX, reserved)",
+			past:     "Instance usage (Linux/UNIX, on-demand)",
+			expected: "Instance usage (Linux/UNIX, on-demand → reserved)",
+		},
+		{
+			current:  "Instance usage (Linux/UNIX, reserved, t3.medium)",
+			past:     "Instance usage (Linux/UNIX, on-demand, t3.small)",
+			expected: "Instance usage (Linux/UNIX, on-demand → reserved, t3.small → t3.medium)",
+		},
+	}
```

Comment:
It's probably not actually possible, but it would be good to add a case where the number of labels in the () differ, just to make sure we don't panic if that ever happens.  Maybe something like:
```go
		{
			current:  "Instance usage (Linux/UNIX, reserved, t3.medium)",
			past:     "Instance usage (Linux/UNIX, t3.small)",
			expected: "Instance usage (Linux/UNIX, reserved, t3.medium)",
		},
```

Reply:
Good point, added these two test cases:
```go
		{
			current:  "Instance usage (Linux/UNIX, reserved, t3.medium)",
			past:     "Instance usage (Linux/UNIX, on-demand)",
			expected: "Instance usage (Linux/UNIX, on-demand → reserved, t3.medium)",
		},
		{
			current:  "Instance usage (Linux/UNIX, reserved)",
			past:     "Instance usage (Linux/UNIX, on-demand, t3.small)",
			expected: "Instance usage (Linux/UNIX, on-demand → reserved, t3.small)",
		},
```
Although honestly not sure what the expected behaviour should be here though. Maybe we should just return the current value if the label counts don't match?

Reply:
I guess the main thing is that it doesn't blow up so I don't think we should go to , but it does seem a little wierd that we're preserving the "t3.small" in the second case.  Maybe if the length changes diff the whole label set like "Instance usage (Linux/UNIX, on-demand, t3.small) ->  (Linux/UNIX, reserved)"?

Reply:
Good idea, have updated to do this.